spring-beans/src/main/java/org/springframework/beans/AbstractNestablePropertyAccessor.java

```diff
@@ -779,7 +778,7 @@ private void growCollectionIfNecessary(Collection<Object> collection, int index,
 		}
 		int size = collection.size();
 		if (index >= size && index < this.autoGrowCollectionLimit) {
-			Class<?> elementType = ph.getResolvableType().getNested(nestingLevel).asCollection().resolveGeneric();
+			Class<?> elementType = ph.toTypeDescriptor().getResolvableType().getNested(nestingLevel).asCollection().resolveGeneric();
```

Comment:
We haven't triaged the incoming issue yet and I can't say for sure if that change is needed but one way to make that more explicit would be to write some tests that fail without your change and pass with that change. We'd need tests for the issue you've raised.

Reply:
Hi, I submitted a test case
[https://github.com/spring-projects/spring-framework/pull/23944/commits/51ec366d870b97cfd648fdc18f2f25a0383064a6](https://github.com/spring-projects/spring-framework/pull/23944/commits/51ec366d870b97cfd648fdc18f2f25a0383064a6)